src/hooks/useMessageValidations.ts

```diff
@@ -0,0 +1,118 @@
+import { useT } from "@transifex/react";
+
+interface IntersectionInfo {
+  intersectSmaller: boolean;
+  percentage: number;
+  poly_name: string | null;
+}
+
+interface ProjectGoalInfo {
+  sum_area: number;
+  percentage: number;
+  total_area_project: number;
+}
+
+interface ExtraInfoItem {
+  exists: boolean;
+  field: string;
+  error?: string;
+}
+
+export const useMessageValidators = () => {
+  const t = useT();
+
+  const getIntersectionMessages = (extraInfo: any): string[] => {
+    if (!extraInfo) return [];
+    try {
+      const infoArray: IntersectionInfo[] = JSON.parse(extraInfo);
+      return infoArray.map(({ intersectSmaller, percentage, poly_name }: IntersectionInfo) => {
+        return intersectSmaller
+          ? t("Geometries intersect: approx. {percentage}% of another, smaller polygon ({poly_name})", {
+              percentage,
+              poly_name: poly_name || "Unnamed Polygon"
+            })
+          : t("Geometries intersect: approx. {percentage}% of this polygon is intersected by {poly_name}", {
+              percentage,
+              poly_name: poly_name || "Unnamed Polygon"
+            });
+      });
+    } catch (error) {
+      console.error(error);
+      return [t("Error parsing extra info.")];
+    }
+  };
+  const getDataMessage = (extraInfo: string | undefined) => {
+    if (!extraInfo) return [];
+
+    try {
+      const fieldsToValidate: any = {
+        poly_name: "Polygon Name",
+        plantstart: "Plant Start",
+        plantend: "Plant End",
+        practice: "Restoration Practice",
+        target_sys: "Target Land Use System",
+        distr: "Tree Distribution",
+        num_trees: "Number of Trees"
+      };
+      const infoArray: ExtraInfoItem[] = JSON.parse(extraInfo);
+      return infoArray
+        .map(info => {
+          if (info.exists === false) {
+            return t("{field} is missing.", { field: fieldsToValidate[info.field] });
+          } else if (info.exists === true && info.field === "target_sys") {
```

Comment:
just use `info.exists` as part of the condition.

Reply:
done